pkg/cloud/services/network/vpc.go

```diff
@@ -54,6 +54,11 @@ func (s *Service) reconcileVPC() error {
 			return errors.Wrap(err, ".spec.vpc.id is set but VPC resource is missing in AWS; failed to describe VPC resources. (might be in creation process)")
 		}
 
+		actualIPv6Enabled := (vpc.IPv6 != nil)
+		if s.scope.VPC().IsIPv6Enabled() != actualIPv6Enabled {
+			return errors.Errorf("IPv6 support of found unmanaged VPC %s differs from desired spec. Changing IP family is currently not supported.", vpc.ID)
+		}
```

Comment:
This seems a net-new check (repeated in the other function as well); should we remove this for now and add it in a different PR? Generally these checks would be best in a webhook at Update time

Reply:
The webhook already checks that IPv6 can't be toggled later:

https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/a1918aa166c219fbd64d32b30d01696db3fd79f7/controlplane/eks/api/v1beta2/awsmanagedcontrolplane_webhook.go#L162-L165

IIRC, my change was to get a clear error message in unit tests, since they can simulate mismatching situations such as someone manually having turned on IPv6 via AWS Console. In such scenarios, the webhook wouldn't be called, but we still want an error instead of being silent about not reconciling the difference. I don't think this would be a user-facing breakage since normally, the reconciled state (here: IPv6 support on/off in the VPC) matches the spec.

Reply:
Agree with @vincepri here, it's better to do it in separate PR.

Reply:
I removed the 2 occurrences of the check, and the new test case for that. I'll take note to re-add this in a separate PR, potentially targeting a later milestone.